### PR TITLE
Security fix 7 -  DoS due to inv-to-send sets 

### PR DESCRIPTION
**Pull request by Naviabheeman**
- PR 27610 from bitcoin is ported
 - unit test for mempool changes from commit ed54696 is also ported

### CODE REVIEW

File: src/test/mempool_tests.cpp [hunk 1]
```diff
@@ -758,4 +758,46 @@ BOOST_AUTO_TEST_CASE(MempoolAncestryTests)
...
+    BOOST_CHECK(!pool.CompareDepthAndScore(tb->GetHashMalFix(), te->GetHashMalFix()));
+    BOOST_CHECK(!pool.CompareDepthAndScore(tc->GetHashMalFix(), te->GetHashMalFix()));
+    BOOST_CHECK(!pool.CompareDepthAndScore(td->GetHashMalFix(), te->GetHashMalFix()));
+    BOOST_CHECK(!pool.CompareDepthAndScore(te->GetHashMalFix(), te->GetHashMalFix()));
+    BOOST_CHECK(!pool.CompareDepthAndScore(tf->GetHashMalFix(), te->GetHashMalFix()));
+    BOOST_CHECK(pool.CompareDepthAndScore(tf->GetHashMalFix(), ta->GetHashMalFix()));
+    BOOST_CHECK(pool.CompareDepthAndScore(tf->GetHashMalFix(), tb->GetHashMalFix()));
+    BOOST_CHECK(pool.CompareDepthAndScore(tf->GetHashMalFix(), tc->GetHashMalFix()));
+    BOOST_CHECK(pool.CompareDepthAndScore(tf->GetHashMalFix(), td->GetHashMalFix()));
+}
```
Comment (azuchi) [inline src/test/mempool_tests.cpp:802]: Is this test related to this PR?

Comment (Naviabheeman) [reply src/test/mempool_tests.cpp:802]: yes, I got its link from one of the conversations in the PR : https://github.com/bitcoin/bitcoin/pull/27610/files#r1191209626

**Review by azuchi — APPROVED** [review]
